pushnotifications/src/main/java/com/pusher/pushnotifications/reporting/ReportingJobService.kt

```diff
@@ -23,7 +23,7 @@ data class PusherMetadata(
     get() = _hasData ?: false
 }
 
-class ReportingJobService: JobService() {
+open class ReportingJobService: JobService() {
```

Comment:
Production code should not be compromised for the purpose of the test code. Well designed code is naturally testable. When it comes to this class though making it open is not a biggie though but it would be good to at least document why it's marked as such and that actually it's not designed for extension. Even better would be to extract that logic and have a test mostly testing our code as opposed to stretching to testing and relying on deprecated Firebase JobDispatcher and all the other 3rd party dependencies which make the test way more complicated (and more costly in the terms of future maintenance).